src/main/java/org/broadinstitute/consent/http/models/InstitutionDomainMap.java

```diff
@@ -18,4 +18,13 @@ public void setInstitutionDomainMap(Map<String, Set<String>> institutionDomainMa
   public Set<String> getDomainsForInstitution(String institution) {
     return institutionDomainMap.get(institution);
   }
+
+  public String getInstitutionForEmail(String email) {
+    String domain = email.substring(email.indexOf('@') + 1);
+    return institutionDomainMap.entrySet().stream()
```

Comment:
If this structure is inconvenient for searching, should we make an additional (reverse) mapping internally from `domain` -> `institution`?

Reply:
Since the structure isn't persisted across API calls there's no benefit in creating a reverse map, as the time to create it will match the linear search done here.

If we decide to change this to use an in-memory cache, I agree that creating a reverse map would be worthwile.